jwst/resample/resample_utils.py

```diff
@@ -140,33 +140,35 @@ def _reproject(x, y):
 
 
 def build_driz_weight(model, weight_type=None, good_bits=None):
-    """ Create input weighting image
+    """Create a weight map for use by drizzle
     """
     dqmask = build_mask(model.dq, good_bits)
     exptime = model.meta.exposure.exposure_time
 
-    if weight_type == 'error':
-        err_model = np.nan_to_num(model.err)
-        inwht = (exptime / err_model)**2 * dqmask
-        log.debug("DEBUG weight mask: {} {}".format(type(inwht), np.sum(inwht)))
-    # elif weight_type == 'ivm':
-    #     _inwht = img.buildIVMmask(chip._chip,dqarr,pix_ratio)
+    if weight_type == 'ivm':
+        with np.errstate(divide="ignore", invalid="ignore"):
```

Comment:
Ooh, nice. I have to remember to use this in the future.

Reply:
Yes, useful!  Though it has it's pitfalls. I only did it after I made sure the arrays looked how I expected them to, as the warning messages were telling me something.  For instance, dividing by zero variance produced `np.inf`, but the original code here that used `np.nan_to_num` then turned that `np.inf` into:

```
In [2]: np.nan_to_num(np.inf)                                                                                           
Out[2]: 1.7976931348623157e+308
```

which is clearly wrong.  Hence the call below to `np.isfinite`.